lib/redix/pubsub.ex

```diff
@@ -329,6 +329,17 @@ defmodule Redix.PubSub do
     :gen_statem.call(conn, {:subscribe, List.wrap(channels), subscriber})
   end
 
+  @doc """
+  Sends a ping and waits for a reply from the server.
+
+  Upon a successful reply from the server, :ok will be returned. If no reply is received within 5 seconds, :error will be returned.
+
+  """
+  @spec ping(connection()) :: :ok | :error
+  def ping(conn) do
```

Comment:
We need to make the timeout here configurable, and not hardcoded to 5 seconds. Do this:

```elixir
def ping(conn, timeout \\ 5000) when is_integer(timeout) and timeout >= 0 do
```

and trickle that down to the implementation.